delivery_cttexpress/models/delivery_carrier.py

```diff
@@ -253,6 +262,7 @@ def cttexpress_get_label(self, reference):
         if not reference:
             return False
         ctt_request = self._ctt_request()
+        time.sleep(self.cttexpress_label_delay)
```

Comment:
@chienandalu Don't do this.

Instead, I suggest you to use the session timeout argument of the zeep client.

See : https://docs.python-zeep.org/en/master/transport.html

Reply:
It's not for that. The problem is that CTT is returning immediately the response, but preparing on background the label, so asking very soon for it will return a "404" (well, the equivalent). This is the only method.

Reply:
> This is the only method.

Nope.

Further, in the get_documents_multi, the client - if timeout defined correctly - will behave as expected without sleep() needs: https://github.com/Tecnativa/delivery-carrier/blob/e55397fda15d1d2845d760f23f6e7fd8df262875/delivery_cttexpress/models/cttexpress_request.py#L181

Reply:
The better will be to implement a 'retry' mechanism for the '404' errors. As you cannot do a back and forth with the sleep timeout for label...

Reply:
> Further, in the get_documents_multi, the client - if timeout defined correctly - will behave as expected without sleep() needs

Yes, but when that method is invoked you already have a shipping reference...

And that's the issue precisely. As CTT explained us their system is quite clunky (well that's our conclusion :smile: ).

The main issue here is that the backend part for the label generation has some kind of cron and you can't expect to have the label inmidiately. If I remember well, there's no error, you'll only get an empty response.

Could we do a retry mechanism and retry forever? or after a `n` number o loops? Should we wait a little between loops to avoid issues with hundreds of successive calls against a weakly engineered API...?  😏️

Reply:
I think it's enough in this case. It's true that is not the most elegant thing, but this is not active by default and we can't invest more here.